### PR TITLE
feat(atomic): add interactive a11y test helpers and reporter support

### PR DESCRIPTION
**Pull request by y-lakhdar**
## Context

This PR adds **interactive accessibility test helpers** and extends the `VitestA11yReporter` to capture their results — including **pass/fail status** per test and per criterion. It's the foundation for verifying WCAG criteria that axe-core cannot evaluate — keyboard navigation, ARIA state management, focus order, and live region announcements.

The helpers live in `packages/atomic/storybook-utils/a11y/` and are designed to be called from Storybook `play` functions. Each helper targets a specific WAI-ARIA interaction pattern, adapted to Atomic's actual component implementations (which sometimes diverge from generic APG patterns).

### Interactive a11y test helpers (`packages/atomic/storybook-utils/a11y/`)

Eight helpers, each covering a distinct interaction pattern:

| Helper | Pattern | WCAG criteria covered |
|--------|---------|----------------------|
| **`collapsible.ts`** | Expand/collapse (smart snippet, generated answer) | 2.1.1, 4.1.2 |
| **`combobox.ts`** | Search box with suggestions (textarea + aria-haspopup) | 2.1.1, 2.1.2, 2.4.3, 2.4.7, 4.1.2 |
| **`dialog.ts`** | Modal dialog (refine modal, quickview) | 2.1.2, 2.4.3, 2.4.7, 4.1.2 |
| **`disclosure.ts`** | Toggle button → modal (refine toggle) | 2.1.1, 4.1.2 |
| **`listbox.ts`** | Facet selection controls (aria-pressed buttons, checkboxes) | 2.1.1, 2.4.3, 4.1.2 |
| **`live-region.ts`** | Live region announcements (query summary, notifications) | 4.1.3 |
| **`navigation.ts`** | Navigation landmarks and interactive elements (pager, sort) | 2.1.1, 2.4.3, 4.1.2 |
| **`tab.ts`** | Tab manager (role="list" with aria-current, not APG tablist) | 2.1.1, 2.4.3, 2.4.7, 4.1.2 |

#### Failure safety: try/catch/finally pattern

All 8 helpers wrap their `step()` calls in `try/catch/finally` to ensure `addReport()` **always fires** — even when assertions fail and the test throws. Previously, a failing assertion would throw before `addReport()` was called, making interactive test failures completely invisible in the report.

```typescript
let status: 'passed' | 'failed' = 'passed';
try {
  // all step() calls here
} catch (error) {
  status = 'failed';
  throw error;  // re-throw to preserve Vitest failure reporting
} finally {
  context.reporting?.addReport?.({ type: 'a11y-interactive', version: 1, status, result: { criteriaCovered: [...] } });
}
```

### Reporter status capture (`packages/atomic-a11y/src/reporter/`)

The reporter now tracks **pass/fail status** of interactive tests, not just coverage. This enables downstream consumers (OpenACR converter) to determine WCAG conformance from interactive test results.

#### Cross-check logic (vitest-a11y-reporter.ts)

The reporter cross-checks two signals for each interactive test:
1. `interactiveReport.status` — what the helper reported (`'passed'` | `'failed'` | `'warning'`)
2. `testCase.result().state` — Vitest's test outcome (`'passed'` | `'failed'` | `'skipped'` | `'pending'`)

If Vitest says the test **failed** but the report says passed (edge case: throw happened between status assignment and catch), the reporter forces `'failed'`. The `'warning'` status counts toward coverage but NOT toward pass/fail conformance.

#### New fields

**Component accumulator** (`reporter-utils.ts`):
- `passedCount`, `failedCount` — count of passed/failed interactive tests
- `passedCriteria: Set<string>`, `failedCriteria: Set<string>` — criteria with passing/failing tests

**Component level** — `A11yComponentReport.interactive?`:
```json
{
  "name": "atomic-tab",
  "automated": { "violations": 0, "passes": 18, "criteriaCovered": ["1.1.1", "..."] },
  "interactive": {
    "criteriaCovered": ["2.1.1", "2.4.3", "2.4.7", "4.1.2"],
    "testCount": 1,
    "passedCount": 1,
    "failedCount": 0,
    "failedCriteria": []
  }
}
```

**Criteria level** — `A11yCriterionReport.interactiveStatus?`:
```json
{
  "id": "2.1.1",
  "name": "Keyboard",
  "automatedCoverage": true,
  "interactiveCoverage": true,
  "interactiveStatus": "passed",
  "manualVerified": false
}
```

- `'passed'` — all components' interactive tests passed for this criterion
- `'failed'` — all components' interactive tests failed for this criterion
- `'mixed'` — some components passed and others failed for this criterion
- Omitted when no interactive tests cover this criterion

**Summary level** — `A11ySummary.interactivePassRate`:
```json
{
  "automatedCoverage": "38%",
  "interactiveCoverage": "7%",
  "interactivePassRate": "100%",
  "manualCoverage": "0%"
}
```

Percentage of interactively-covered criteria that have `interactiveStatus === 'passed'`.

### Data flow

```
Story play() → helper wraps steps in try/catch/finally
  → addReport({type: 'a11y-interactive', status: 'passed'|'failed', criteriaCovered: [...]})
  → Vitest passes through meta.reports
  → VitestA11yReporter.onTestCaseResult():
    - Cross-checks testCase.result().state with interactiveReport.status
    - Effective status: test failed → force 'failed'; warning → coverage only
    - Populates passedCount/failedCount/passedCriteria/failedCriteria
  → buildA11yReport():
    - Serializes interactive section per component
    - Computes interactiveStatus per criterion (passed/failed/mixed)
  → createSummary() computes interactivePassRate
  → Output: a11y-report.json
```

## PR Chain

| # | PR | Branch | Description |
|---|-----|--------|-------------|
| 1 | #7111 | `feat/a11y-package-scaffold` | Package scaffolding |
| 2 | #7122 | `feat/a11y-shared-foundation` | Shared types, constants, utilities |
| 3 | #7123 | `feat/a11y-reporter` | VitestA11yReporter |
| **—** | **#7285** | **`feat/a11y-interactive`** | **Interactive a11y test helpers + reporter support ← this PR** |
| — | #7286 | `feat/a11y-interactive-story-implement` | Wire interactive tests into component stories |
| 4 | #7124 | `feat/a11y-openacr` | OpenACR converter modules |
| 5a | #7187 | `feat/a11y-delta-workflow` | Delta audit validation + shared foundation |
| 5b | #7188 | `feat/a11y-browser-llm` | Browser capture + LLM client |
| 5c | #7189 | `feat/a11y-audit-orchestrator` | AI WCAG audit orchestrator + prompts |

KIT-5471

### CODE REVIEW

**Review by alexprudhomme — APPROVED** [review]

File: packages/atomic/storybook-utils/a11y/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export {
+  assertExpandCollapse,
```
[the remark this answers is not in the document]
Comment (y-lakhdar) [reply packages/atomic/storybook-utils/a11y/index.ts]: good catch. old code. removed
